synapse/models/inet.py

```diff
@@ -3306,6 +3464,267 @@ def getModelDefs(self):
                             'ro': True,
                             'doc': 'The x509 certificate sent by the client.'})
                     )),
+                    ('inet:service:platform', {}, (
+
+                        ('url', ('inet:url', {}), {
+                            'ex': 'https://twitter.com',
+                            'doc': 'The primary URL of the platform.'}),
+
+                        ('name', ('str', {'onespace': True, 'lower': True}), {
+                            'ex': 'twitter',
+                            'doc': 'A friendly name for the platform.'}),
+
+                        ('provider', ('ou:org', {}), {
+                            'doc': 'The organization which operates the platform.'}),
+
+                        ('provider:name', ('ou:name', {}), {
+                            'doc': 'The name of the organization which operates the platform.'}),
+                    )),
+
+                    ('inet:service:instance', {}, (
+
+                        ('id', ('str', {'strip': True}), {
+                            'ex': 'B8ZS2',
+                            'doc': 'A platform specific ID to identify the service instance.'}),
+
+                        ('platform', ('inet:service:platform', {}), {
+                            'ex': '{ inet:service:platform:name=slack }',
+                            'doc': 'The platform which defines the service instance.'}),
+
+                        ('url', ('inet:url', {}), {
+                            'ex': 'https://v.vtx.lk/slack',
+                            'doc': 'The primary URL which identifies the service instance.'}),
+
+                        ('name', ('str', {'lower': True, 'onespace': True}), {
+                            'ex': 'synapse users slack',
+                            'doc': 'The name of the service instance.'}),
+                    )),
+
+                    ('inet:service:account', {}, (
+
+                        ('id', ('str', {'strip': True}), {
+                            'doc': 'A platform specific ID used to identify the account.'}),
+
+                        ('user', ('inet:user', {}), {
+                            'doc': 'The current user name of the account.'}),
+
+                        ('email', ('inet:email', {}), {
+                            'doc': 'The current email address associated with the account.'}),
+
+                        ('profile', ('ps:contact', {}), {
+                            'doc': 'Profile details associated with the account.'}),
+                    )),
+
+                    ('inet:service:group', {}, (
```

Comment:
Should this have :platform and :instance props?

Reply:
It has them via the `inet:service:object` interface